CIP-0164/README.md

```diff
@@ -0,0 +1,2774 @@
+| Resilience | Adversarial stake, $\eta_\text{adversary}(s)$      | Fractional loss in throughput due to adversial stake of $s$                                           |
+
+<em>Table 5: Performance Metrics</em>
+
+</div>
+
+**_Spatial efficiency:_** Leios necessarily imposes some disk overhead beyond
+the raw bytes needed to store transactions themselves. This overhead includes
+the EBs and RBs associated with storing transactions. The spatial efficiency
+metric is defined as the ratio of the total bytes of transactions included in
+the ledger to the total persistent storage required by the protocol.
+
+$$
+\epsilon_\text{spatial} = \frac{\text{total bytes of transactions included in the ledger}}{\text{total bytes of EBs and RBs}}
+$$
+
+**_Temporal efficiency:_** As is true for Praos, there is a delay between
+submitting a transaction and its being included in the ledger and there is a
+finite chance that it never is included in the ledger. Before a transaction is
+endorsed, it must be validated and placed in the memory pool. It is cleanest to
+measure the time from the transaction reaching the local memory pool of the node
+where it was submitted to the time when it is included in the ledger, via a
+Praos block. The same metric applies both to Praos and to Leios. In aggregate,
+we measure the temporal efficiency as the fraction of transactions that reach
+the ledger, as function of the number of slots elapsed. The quantity
+$\epsilon_\text{temporal}(\infty)$ is the fraction of submitted transactions
+that ever reach the ledger.
+
+$$
+\epsilon_\text{temporal}(s) = \text{fraction of transactions included in the ledger within } s \text{ slots of their inclusion in a local memory pool}
+$$
+
+**_Network efficiency:_** Effective utilization of the network can be
+characterized by the ratio of bytes of transactions reaching the ledger to the
+average network traffic per node. (This could also be computed individually for
+each node and used as a local metric.)
+
+$$
+\epsilon_\text{network} = \frac{(\text{bytes of valid transactions reaching the ledger}) \cdot (\text{number of nodes in the network})}{\text{total bytes of network traffic}}
+$$
+
+**_TX inclusion:_** In Leios, it is possible that a transaction might have to
+wait for multiple EB production opportunities before being included in an EB.
+The characteristic time for such inclusion in an EB depends on the EB production
+rate and mempool management. This is correlated with how long the transaction
+waits in the memory pool before being selected for inclusion.
+
+$$
+\tau_\text{inclusion} = \text{mean number of slots for a transaction to be included in any EB}
+$$
+
+**_Voting failure:_** An unlucky set of VRF evaluations might result in
+insufficient voters being selected in a given pipeline, thus making it
+impossible to certify an EB in that pipeline.
```

Comment:
Does the *pipeline* terminology correspond to linear Leios, or is a remnant from the Research/Full version?